crates/api_models/src/gsm.rs

```diff
@@ -26,6 +26,67 @@ pub struct GsmCreateRequest {
     pub unified_code: Option<String>,
     /// error message unified across the connectors
     pub unified_message: Option<String>,
+    /// category in which error belongs to
+    pub error_category: Option<ErrorCategory>,
+    /// sub-category in which error belongs to
+    pub error_sub_category: Option<ErrorSubCategory>,
+}
+
+#[derive(
+    Clone,
+    Copy,
+    Debug,
+    strum::Display,
+    PartialEq,
+    Eq,
+    serde::Serialize,
+    serde::Deserialize,
+    strum::EnumString,
+    ToSchema,
+)]
+#[serde(rename_all = "snake_case")]
+#[strum(serialize_all = "snake_case")]
+pub enum ErrorCategory {
+    IssuerDecline,
```

Comment:
We already have `ConnectorErrorType` which is used for different purpose in auto retries flow, and has the following values
`UserError, BusinessError, TechnicalError, UnknownError`

Should we follow the same here? @Chethan-rao 
cc: @jarnura